src/auto-evo/RunResults.cs

```diff
@@ -811,6 +826,144 @@ void OutputPopulationForPatch(Species species, Patch patch, long population)
             return builder;
         }
 
+        public void LogResultsToTimeline(GameWorld world, List<ExternalEffect> effects = null)
+        {
+            var newSpecies = GetNewSpecies();
+
+            foreach (var patch in world.Map.Patches.Values)
+            {
+                foreach (var species in patch.SpeciesInPatch.Keys)
+                {
+                    var hasPopulationBeforeMigration = GetSpeciesPopulationsByPatch(species, true, true)
+                        .TryGetValue(patch, out var finalPatchPopulation);
+
+                    var globalPopulation = GetGlobalPopulation(species, true, true);
+                    var previousGlobalPopulation = world.Map.GetSpeciesGlobalPopulation(species);
+
+                    var previousPatchPopulation = patch.GetSpeciesPopulation(species);
+
+                    // Adjust final patch population by external effects
+                    if (effects != null)
+                    {
+                        foreach (var effect in effects)
+                        {
+                            if (effect.Species == species)
+                            {
+                                finalPatchPopulation += effect.Constant + (long)(effect.Species.Population *
+                                    effect.Coefficient) - effect.Species.Population;
+                            }
+                        }
+                    }
+
+                    if (globalPopulation <= 0)
+                    {
+                        // TODO: see https://github.com/Revolutionary-Games/Thrive/issues/2958
+                        LogEventGloballyAndLocally(world, patch, new LocalizedString(
+                                "TIMELINE_SPECIES_EXTINCT", species.FormattedName),
+                            species.PlayerSpecies, "extinction.png");
+                    }
+                    else
+                    {
+                        if (hasPopulationBeforeMigration)
+                        {
+                            if (finalPatchPopulation > previousPatchPopulation)
+                            {
+                                patch.LogEvent(new LocalizedString("TIMELINE_SPECIES_POPULATION_INCREASE",
+                                        species.FormattedName, ((double)finalPatchPopulation).FormatNumber()),
```

Comment:
Why are the population numbers as doubles here? They can't be fractional so why don't we just include the raw number?

Reply:
Have no clue how to make `FormatNumber` takes in any numeric data types. Any suggestions?

Reply:
Does it need to be formatted? In the auto-evo results the longs are just shown as-is.

Reply:
Yes, for easier digestible numbers, otherwise it can be hard to understand the long string of numbers added on top of columns of events.

Reply:
I think there's no other way than to add an overload of the format method for longs. I had to do that for one place where I couldn't have a general "any number" generic. I think that's a C# limitation we just need to work around.